### ISSUE TITLE
PropertyNotSetInConstructor reported when it shouldn't 

### ISSUE THREAD

**Issue by zmitic** [vimeo/psalm#3266]
https://psalm.dev/r/87598d9ebb

I just downloaded dev-master and it reports this error; this error didn't happen before.

Example provided is simplified, real usage has more to this.

**Comment by someniatko** [comment]
@zmitic This is not a false-positive. Consider this case:

```php
class TrollPostalCode extends PostalCode
{
    public function setCode(string $code): void
    {
        // do nothing
    }
}
```

**Comment by zmitic** [comment]
@someniatko No, this worked before i.e. psalm analyzed ``setCode()`` method and detected assignment.

Version 3.11.2, LSP plugin:

![image](https://user-images.githubusercontent.com/1964158/80610751-0253bd00-8a3a-11ea-9c1a-cd9d9a7a0ef4.png)

Error shown is the same, started when I commented assignment.

**Comment by someniatko** [comment]
@zmitic this worked before, because Psalm did not detect such case before :)
However, still, this is not a false-positive.

Your setter method must be unoverridable - either final or private.

**Comment by muglug** [comment]
Yeah, this is a change in behaviour added to fix https://github.com/vimeo/psalm/issues/3201. Adding `final` to your public method declaration will fix the bug: https://psalm.dev/r/db13fbc3ab
